app/src/ui/open-pull-request/pull-request-files-changed.tsx

```diff
@@ -70,12 +70,71 @@ export class PullRequestFilesChanged extends React.Component<
   IPullRequestFilesChangedProps,
   IPullRequestFilesChangedState
 > {
+  private fileDiffContainer = React.createRef<HTMLDivElement>()
+
   public constructor(props: IPullRequestFilesChangedProps) {
     super(props)
 
     this.state = { showSideBySideDiff: props.showSideBySideDiff }
   }
 
+  public componentDidMount() {
+    this.setfileDiffContainerHeight()
+    window.addEventListener('resize', this.setfileDiffContainerHeight)
+  }
+
+  public componentWillUnmount() {
+    window.removeEventListener('resize', this.setfileDiffContainerHeight)
+  }
+
+  private getPullRequestHeaderHeight() {
+    const query = document.querySelectorAll('.open-pull-request .dialog-header')
+    const elHeight = query.length > 0 ? query.item(0).clientHeight : 0
+    return elHeight > 0 ? elHeight : 87
+  }
+
+  private getPullRequestFillDiffHeaderHeight() {
+    const query = document.querySelectorAll(
+      '.open-pull-request .files-changed-header'
+    )
+    const elHeight = query.length > 0 ? query.item(0).clientHeight : 0
+    return elHeight > 0 ? elHeight : 39
+  }
+
+  private getPullRequestFooterHeight() {
+    const query = document.querySelectorAll('.open-pull-request .dialog-footer')
+    const elHeight = query.length > 0 ? query.item(0).clientHeight : 0
+    return elHeight > 0 ? elHeight : 67
+  }
+
+  private getDiffContainerHeightOffset() {
+    const dialogMargin = 100
```

Comment:
I do not know exactly why.. but need at least 80 or the footer starts clipping on a mac (have not tested windows yet)

Reply:
Figured out that the app title bar needed to be part of the calculation as well because the dialog height is limited to the `desktop-app-contents` which is the `window.innerHeight` minus the `desktop-app-title-bar` height. Therefore, the dialog will always be guaranteed a minimum margin of the height of the title-bar since the dialog is absolutely positioned and margin set to auto to center it.